index.js

```diff
@@ -0,0 +1,192 @@
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+const bestReleases = (releasesArray) => {
+  const selectedReleases = [];
+  releasesArray.forEach((obj) =>
+    obj.rating[0] === 5.0 ? selectedReleases.push(obj.id) : null
+  );
+  return selectedReleases;
+};
+
+// Task 4
+/* С помощью функций map, reduce, вывести url у которого самая большая площадь */
+
+const boxarts = [
+  {
+    width: 200,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture200.jpg",
+  },
+  {
+    width: 150,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture150.jpg",
+  },
+  {
+    width: 300,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture300.jpg",
+  },
+  {
+    width: 425,
+    height: 150,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture425.jpg",
+  },
+];
+
+const getBiggestArea = (boxArray) => {
+  return boxArray.reduce((biggestObj, obj) => {
+    biggestObj =
+      biggestObj.width * biggestObj.height > obj.width * obj.height
+        ? biggestObj
+        : obj;
+    return biggestObj;
+  }).url;
+};
+
+// Task 5
+/* Написать функцию, которая принимает в себя массив из числовых и строчных-числовых значений. Данная ф-ция должна вернуть массив, в которос числовые значения исходного массива стали строчными, а строчные - числовыми */
+
+const convertValues = (valArray) =>
+  valArray.map((value) =>
+    typeof value === "string" ? Number(value) : String(value)
+  );
+
+// Task 6
+/* Написать функцию, которая принимает в себя 2 массива, а возвращает один, состоящий из 2ух, которые пришли в нее */
+
+const joinArrays = (arr1, arr2) => {
+  // Check lenght of each array, add values (do push) to the biggest array to decrease number of iterations for forEach
+  if (arr1.length > arr2.length) {
+    arr2.forEach((val) => arr1.push(val));
+    return arr1;
+  } else {
+    arr1.forEach((val) => arr2.push(val));
+    return arr2;
+  }
+};
+
+// Task 7
+/* Написать функцию, которая принимает следующие значения: первым аргументом - строковое значение, второым аргументом - допустимое кол-во символов.
+Если длина строки больше допустимого кол-ва символов, строка должна обрезаться, а в конец добавляться многоточие */
+
+const cutString = (string, symbols) => {
+  if (!((typeof string === "string") & (typeof symbols === "number"))) {
+    return "Error, the first argument should be of type string and 2nd - of type integer";
+  }
+
+  let stringAsArray = string.split("");
+  if (stringAsArray.length > symbols) {
+    stringAsArray = stringAsArray.slice(0, symbols);
+    return `${stringAsArray.join("")}...`;
+  } else {
+    return string;
+  }
```

Comment:
Strings also have a property `length`. You don't need to convert it to an array.  try to simplify your solution

Reply:
fixed